mindsdb/integrations/handlers/merlion_handler/merlion_handler.py

```diff
@@ -0,0 +1,224 @@
+    return not name in type_class._member_names_
+
+
+def enum_to_str(type_class: Enum) -> str:
+    all = []
+    for element in type_class:
+        all.append(element.name)
+    return "|".join(all)
+
+
+def to_ts_dataframe(df: pd.DataFrame, time_col=None) -> str:
+    columns = list(df.columns.values)
+    if time_col is not None :
+        if time_col not in columns:
+            raise Exception("invalid column name: " + time_col)
+        if df[time_col].dtype != np.datetime64:
+            try:
+                df.index = pd.to_datetime(df[time_col])
+            except Exception as e:
+                raise Exception("can not convert column to datetime: " + time_col + " " + str(e))
+    else:
+        datetime_cols = list(df.select_dtypes(include=["datetime"]).columns.values)
+        if len(datetime_cols) > 0:
+            time_col = datetime_cols[0]
+            df.index = pd.to_datetime(df[time_col])
+        if time_col is None:
+            raise Exception("can not find datetime column for time series")
+    df.drop(columns=[time_col], inplace=True)
+    return time_col
+
+
+# remove ` from column, the result is (mapping, back_mapping)
+# this function is a little strange, but I think it's important to keep
+# the raw column names, maybe ` is a trick while parsing sql? maybe ...
+def get_column_name_format_mapping(df: pd.DataFrame) -> (dict, dict):
+    column_names = list(df.columns.values)
+    mapping = {}
+    back_mapping = {}
+    for column_name in column_names:
+        if column_name.find("`") == -1:
+            continue
+        formatted_name = column_name.replace("`", "")
+        mapping[column_name] = formatted_name
+        back_mapping[formatted_name] = column_name
+    return mapping, back_mapping
+
+
+class MerlionHandler(BaseMLEngine):
+    name = 'merlion'
+
+    ARG_USING_TASK = "task"
+    ARG_USING_MODEL_TYPE = "model_type"
+    ARG_USING_TIME_COLUMN = "time_column"
+    ARG_HORIZON = "horizon"
+    ARG_WINDOW = "window"
+    ARG_TARGET = "target" # only be used to persist args to args.json
+    ARG_COLUMN_SEQUENCE = "column_sequence"  # only be used to persist args to args.json
+
+    KWARGS_DF = "df"
+
+    DEFAULT_MODEL_TYPE = "default"
+    DEFAULT_MAX_PREDICT_STEP = 100
+    DEFAULT_PREDICT_BASE_WINDOW = 10
+
+    PERSISIT_MODEL_FILE_NAME = "merlion_model"
+    PERSISIT_ARGS_KEY_IN_JSON_STORAGE = "args"
+
+    def create(self, target, args=None, **kwargs):
+        df: pd.DataFrame = kwargs.get(self.KWARGS_DF, None)
+        column_name_mapping, column_name_back_mapping = get_column_name_format_mapping(df)
+        df.rename(columns=column_name_mapping, inplace=True)
+
+        # prepare arguments
+        task = args.get(self.ARG_USING_TASK, TaskType.forecast.name)
+        model_type = args.get(self.ARG_USING_MODEL_TYPE, self.DEFAULT_MODEL_TYPE)
+        time_column = args.get(self.ARG_USING_TIME_COLUMN, None)
+        horizon = args.get(self.ARG_HORIZON, self.DEFAULT_MAX_PREDICT_STEP)
```

Comment:
horizon and window are defined in BaseMLEngineExec. So we can use these names as convectional, without special constant. Just args.get('horizon'...)

Reply:
Because I used this string twice， so I think maybe use a constant will be easier to maintain the code.
No problem, I've use string directly to replace the constant.